src/app/aether-slice/slice-edit/slice-edit.component.ts

```diff
@@ -693,7 +693,6 @@ export class SliceEditComponent
                         ? this.filterUpf(site)
                         : [selectedUpf, ...this.filterUpf(site)];
 
-                    this.upfs = [...this.filterUpf(site)];
```

Comment:
this was overwriting the value from above and discarding the currently selected UPF

Reply:
Looks good, that was probably an artifact of some big refactoring but I see it's node on line `694`. Thanks for fixing it.